HW0/P4/P4.py

```diff
@@ -40,37 +45,93 @@
     # Read the observation array and plot it (Part 2)
     #####################
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    s_obs = np.loadtxt('P4_measurements.txt', delimiter=',')
+    x_coords = [row[0]/rx for row in s_obs]
+    y_coords = [row[1]/ry for row in s_obs]
+    z_coords = [row[2]/rz for row in s_obs]
+    ax.plot(x_coords, y_coords, z_coords,'.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+    A = np.matrix([[1, 0, 0, dt, 0, 0],
+                   [0, 1, 0, 0, dt, 0],
+                   [0, 0, 1, 0, 0, dt],
+                   [0, 0, 0, 1-c*dt, 0, 0],
+                   [0, 0, 0, 0, 1-c*dt, 0],
+                   [0, 0, 0, 0, 0, 1-c*dt]])
+    a = np.matrix([0, 0, 0, 0, 0, g * dt])
+    s = np.matrix([0, 0, 2, 15, 3.5, 4.0])
 
     # Initial conditions for s0
     # Compute the rest of sk using Eq (1)
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '-k', label='Blind trajectory')
+    z = np.zeros([K, 6])
+    model = np.asmatrix(z)
+    model[0] = s
+    for i in range(1, K):
+        sk = A * s.transpose() + a.transpose()
+        s = sk.transpose()
+        model[i] = s
+
+    x_coords = [model.item(i, 0) for i in xrange(K)]
+    y_coords = [model.item(i, 1) for i in xrange(K)]
+    z_coords = [model.item(i, 2) for i in xrange(K)]
+    ax.plot(x_coords, y_coords, z_coords, '-k', label='Blind trajectory')
 
     #####################
     # Part 4:
     # Use the Kalman filter for prediction
     #####################
 
-    # B = ?
-    # C = ?
+    B = np.matrix([[bx, 0, 0, 0, 0, 0],
```

Comment:
Can use 
B = np.zeros((6, 6), float)
np.fill_diagonal(B, np.array([bx, by, bz, bvx, bvy, bvz]))
B = np.matrix(B)
to fill in the matrix.